app/lib/actions/katello/repository/fetch_pxe_files.rb

```diff
@@ -19,16 +19,19 @@ def run
                       else
                         SmartProxy.default_capsule!
                       end
-            repo_path = repository.full_path(capsule, true).chomp("/")
 
             os = Redhat.find_or_create_operating_system(repository)
+            medium_provider = Katello::ManagedContentMediumProvider.new(
+              OpenStruct.new(
+                content_source: capsule,
+                kickstart_repository: repository))
 
             ueber_cert = ::Cert::Certs.ueber_cert(repository.organization)
             cert = OpenSSL::X509::Certificate.new(ueber_cert[:cert])
             key = OpenSSL::PKey::RSA.new(ueber_cert[:key])
 
-            Redhat::PXEFILES.each_key do |pxe_file|
-              fetch("#{repo_path}/#{os.url_for_boot(pxe_file)}", cert, key)
+            os.boot_file_sources(medium_provider).values.each do |source_pxe_file|
+              fetch(source_pxe_file, cert, key)
```

Comment:
Open question bought up on IRC, can we just do a HEAD request here?

Reply:
Waiting on feedback from the pulp team.

Reply:
So a HEAD request doesn't cause pulp to immediately download the full file, but within 10 minutes pulp goes to put the file into its final 'content store' and as part of that it fetches the entire file.  So we would have to be okay with a 10 minute delay, which is probably okay.  Your thoughts @parthaa ?

Reply:
Although maybe https://github.com/rest-client/rest-client#streaming-responses  is a better solution.  So it would stream it to disk and not take up additional memory, but then the file would just be deleted.